test/lib/usd/translators/testUsdExportImportRoundtripPreviewSurface.py

```diff
@@ -630,17 +641,35 @@ def ValidateUSDStage(self, stage, allSubsets):
 
         # We export without changes to make sure we get back the modified cube without
         # any extra subsets.
-        reexported_usd_path = os.path.abspath('CubeWithAssignedFaces_reexported.usda')
-        cmds.usdExport(mergeTransformAndShape=True,
-            file=reexported_usd_path,
-            shadingMode='useRegistry',
-            exportDisplayColor=False)
+        if useCollections:
+            reexported_usd_path = os.path.abspath('CubeWithAssignedFaces_reexported_CB.usda')
+            cmds.usdExport(mergeTransformAndShape=True,
+                file=reexported_usd_path,
+                shadingMode='useRegistry',
+                exportDisplayColor=False,
+                exportCollectionBasedBindings=True)
+        else:
+            reexported_usd_path = os.path.abspath('CubeWithAssignedFaces_reexported.usda')
+            cmds.usdExport(mergeTransformAndShape=True,
+                file=reexported_usd_path,
+                shadingMode='useRegistry',
+                exportDisplayColor=False)
         
         stage = Usd.Stage.Open(reexported_usd_path)
 
         # Test that the re-exported stage has the exact same GeomsSubsets as the one we imported:
         ValidateUSDStage(self, stage, allSubsets)
 
+    def testGeomSubsetRoundtrip(self):
+        """Test that GeomSubsets numbers stay under control when rountripping
+           with direct material assignment"""
+        self.CommonGeomSubsetRoundtrip(False)
+
+    def testGeomSubsetRoundtripCollectionBased(self):
+        """Test that GeomSubsets numbers stay under control when rountripping
+           with collection based material assignemnt"""
+        self.CommonGeomSubsetRoundtrip(True)
```

Comment:
The way the material checks are done with `ComputeBoundMaterial` work with both assignment methods. The contents of the USDA files will be slightly different since one will use collections based assignment.